plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeErrorCode.java

```diff
@@ -27,6 +28,7 @@
     DELTA_LAKE_BAD_DATA(2, EXTERNAL),
     DELTA_LAKE_BAD_WRITE(3, EXTERNAL),
     DELTA_LAKE_FILESYSTEM_ERROR(4, EXTERNAL),
+    DELTA_LAKE_UNSUPPORTED_EXPRESSION(5, USER_ERROR),
```

Comment:
I'd argue that this is rather an `EXTERNAL` error.
The Delta table is created by another query engine (atm Trino can't create such tables).

In case Trino will eventually support creating/altering delta tables with  constraints that are not supported, this would qualify for a user error.

Reply:
StandardErrorCode uses USER_ERROR for NOT_SUPPORTED as you may already know. If we think the operation as "users tried to read tables having unsupported expression", it will be USER_ERROR.